### PR TITLE
Fixes for Mingw builds

### PR DESCRIPTION
**Pull request by sourcedelica**
Fixes https://github.com/bincrafters/community/issues/455

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def source(self):
         os.rename("libxml2-{0}".format(self.version), self.source_subfolder)
 
     def config_options(self):
-        if self.settings.os == "Windows":
+        if self.settings.os == "Windows" and self.settings.compiler == "Visual Studio":
```
Comment (SSE4) [inline conanfile.py]: fPIC is not needed for MinGW as well, leave just `if self.settings.os == Windows`

File: conanfile.py [hunk 2]
```diff
@@ -118,3 +114,5 @@ def package_info(self):
         self.cpp_info.libs = tools.collect_libs(self)
         if self.settings.os == "Linux" or self.settings.os == "Macos":
             self.cpp_info.libs.append('m')
+        if self.settings.os == "Windows" and self.settings.compiler != "Visual Studio":
+            self.cpp_info.libs.append('ws2_32')
```
Comment (SSE4) [inline conanfile.py]: also leave just `if self.settings.os == Windows`, MSVC needs ws2_32 as well

**Review by SSE4 — CHANGES_REQUESTED** [review]
your commit also added tar file, please remove it

**Comment by sourcedelica** [comment]
Will do @SSE4

**Comment by sourcedelica** [comment]
Hi @theirix:

Does this warning
```
libxml2/2.9.8@bincrafters/stable: WARN: Warning! Static builds in Windows are unstable
```
Also apply to Mingw?

**Comment by theirix** [comment]
> Hi @theirix:
> 
> Does this warning
> 
> ```
> libxml2/2.9.8@bincrafters/stable: WARN: Warning! Static builds in Windows are unstable
> ```
> Also apply to Mingw?

Don't think so, please try it. I don't remember what exactly crashed in a static build, maybe test_package can help.

**Comment by sourcedelica** [comment]
The test package works ok on both Mingw and VS with static libs (the default).  I'm going to take the warning out if it's ok with everyone.

Another thing I did was use `make install` and create the package from the installation directory.

**Comment by SSE4** [comment]
something is broken for MSVC:
```
  C:\projects\conan-libxml2\test_package\test_package.c(2): fatal error C1083: Cannot open include file: 'libxml/xmlversion.h': No such file or directory [C:\projects\conan-libxml2\test_package\build\bcafe2703ef63fadb78c7e7f02638427c3c4151e\test_package.vcxproj]
```
after looking at recipe, I think it's better to just install straight to the `self.package_folder` (use it as prefix), and avoid useless copying in the `package` method

**Comment by sourcedelica** [comment]
Yes for sure.  I’ll fix that and test on all platforms.

**Comment by sourcedelica** [comment]
Tested using the default configuration + Debug on Mac, Linux, MSVC, Mingw.

Should we add `MINGW_CONFIGURATIONS` to the Appveyor config?  What is the standard Bincrafters `MINGW_CONFIGURATIONS`?

One thing I noticed is that for MSVC it always builds 20 or so test executables and copies them to the prefix on `nmake install`.  There is no way to turn it off using configure or the makefile.  I updated the packaging to delete those:
```
	$(BINDIR)\testAutomata.exe\
	$(BINDIR)\testC14N.exe\
	$(BINDIR)\testDocbook.exe\
	$(BINDIR)\testHTML.exe\
	$(BINDIR)\testReader.exe\
	$(BINDIR)\testRelax.exe\
	$(BINDIR)\testRegexp.exe\
	$(BINDIR)\testModule.exe\
	$(BINDIR)\testSAX.exe\
	$(BINDIR)\testSchemas.exe\
	$(BINDIR)\testURI.exe\
	$(BINDIR)\testXPath.exe\
	$(BINDIR)\runtest.exe\
	$(BINDIR)\runsuite.exe\
	$(BINDIR)\runxmlconf.exe\
	$(BINDIR)\testapi.exe\
	$(BINDIR)\testchar.exe\
	$(BINDIR)\testdict.exe\
	$(BINDIR)\testlimits.exe\
	$(BINDIR)\testrecurse.exe
```

**Comment by sourcedelica** [comment]
Re: The Windows static warning message - `build.py` removes shared builds if MSVC :)  Both the MSVC and Mingw `test_package` run successfull with static.  I am going to remove the warning on the next iteration.

**Comment by SSE4** [comment]
@sourcedelica for MinGW hard to say, e.g. libjpeg has the following:
```
- MINGW_CONFIGURATIONS: '4.9@x86_64@seh@posix, 4.9@x86_64@sjlj@posix, 4.9@x86@sjlj@posix, 4.9@x86@dwarf2@posix, 6@x86@dwarf2@posix, 6@x86_64@seh@posix, 7@x86@dwarf2@posix, 7@x86_64@seh@posix, 8@x86@dwarf2@posix, 8@x86_64@seh@posix'
```

**Comment by sourcedelica** [comment]
It looks like the only way it will build shared on Mingw is if zlib and iconv are also shared.  And there appears to be an issue with the shared build of zlib.  So in addition to MSVC I've also removed the shared version in `build.py` for Mingw.
```
*** Warning: linker path does not have real file for library -lz.
*** I have the capability to make that library automatically link in when
*** you link to this library.  But I can only do this if you have a
*** shared version of the library, which you do not appear to have
*** because I did check the linker path looking for a file starting
*** with libz and none of the candidates passed a file format test
*** using a file magic. Last file checked: C:/Users/eric/.conan/data/mingw_installer/1.0/conan/stable/package/ab30f42c63a1376704be612ea2bd48533323d66a/x86_64-w64-mingw32/lib/libz.a
*** The inter-library dependencies that have been dropped here will be
*** automatically added whenever a program is linked with this library
*** or is declared to -dlopen it.

*** Since this library must not contain undefined symbols,
*** because either the platform does not support them or
*** it was explicitly requested with -no-undefined,
*** libtool will only create a static version of it.
  CCLD     xmllint.exe
  CCLD     xmlcatalog.exe
xmllint.o: In function `xmllintExternalEntityLoader':
C:\users\eric\.conan\data\libxml2\2.9.8\bincrafters\testing\build\e02473d6581fcb5ffad234fe8e2cf8f83a0c34eb\source_subfolder/xmllint.c:307: undefined reference to `__imp_xmlFree'
```
The shared zlib build creates `libzlib.dll.a` so I'm not sure why we get the error above.

**Comment by sourcedelica** [comment]
I experimented with adding `MINGW_CONFIGURATIONS`.  It didn't work because the corresponding  libiconv libraries were not pre-built.  This worked for me on my computer but didn't work on Appveyor because of a 1 hr build time limit.   Also zlib has `MINGW_CONFIGURATIONS: "4.9@x86_64@seh@posix, 5@x86_64@seh@posix, 6@x86_64@seh@posix, 7@x86_64@seh@posix"` which is a smaller set than I tried, so many of the zlib variant were not prebuilt either.  For now let's skip `MINGW_CONFIGURATIONS` and revisit when it is added to libiconv.

I believe this is ready to merge.

**Comment by SSE4** [comment]
okay, great! thank you for your contribution

**Review by SSE4 — APPROVED** [review]
LGTM

**Comment by sourcedelica** [comment]
My pleasure - thanks for the feedback.

**Pull request merged**